src/common/tiglcommonfunctions.cpp

```diff
@@ -1156,6 +1160,125 @@ TopoDS_Wire BuildWireFromEdges(const TopoDS_Shape& edges)
     return result;
 }
 
+opencascade::handle<Geom_BSplineCurve> ApproximateArcOfCircleToRationalBSpline(double radius, double uMin, double uMax,double y_position, double z_position)
+{
+    std::vector<gp_Pnt> ArcPnts;
+    std::vector<double> alpha = Linspace(uMin, uMax, 64);
+    for (double a : alpha){
+        ArcPnts.push_back(gp_Pnt(0., y_position + radius * std::cos(a), z_position + radius* std::sin(a)));
+            }
+    return tigl::CTiglPointsToBSplineInterpolation(ArcPnts).Curve();
+}
+
+
+//gp_Pnt startUpper(0., (-0.5 + radius), 0.5 * heightToWidthRatio);
+//gp_Pnt endUpper(0., (0.5 - radius), 0.5 * heightToWidthRatio);
+//gp_Pnt startRight(0., 0.5, 0.5 * heightToWidthRatio - radius);
+//gp_Pnt endRight(0., 0.5, -0.5 * heightToWidthRatio + radius);
+//gp_Pnt startLower(0., (0.5 - radius), -0.5 * heightToWidthRatio);
+//gp_Pnt endLower(0., (-0.5 + radius), -0.5 * heightToWidthRatio);
+//gp_Pnt startLeft(0., -0.5, -0.5 * heightToWidthRatio + radius);
+//gp_Pnt endLeft(0., -0.5, 0.5 * heightToWidthRatio - radius);
+
+TopoDS_Wire BuildWireRectangle(const double& heightToWidthRatio, const double& cornerRadius)
+{
+    double radius = cornerRadius;
+    if(cornerRadius<0.){
+        radius*= -1;
+    }
+    std::vector<Handle(Geom_BSplineCurve)> curves;
+
+    // build half upper line from gp_points
+    std::vector<gp_Pnt> linePntsUpperRightHalf;
+    std::vector<double> y_UpperRightHalf = Linspace(0.,(0.5-radius),5);
```

Comment:
Yes, thanks for pointing that out. I checked it, this works for all the lines except one, but doesn't matter which one, if I want to use guide curves, at least in my example. I don't understand yet exactly why at least one line requires three points. Otherwise the lofting algorithm doesn't produce a valid shape. I'm wondering whether it has to do with the number of edges of each profile or the point lists that are produced for the lofting. Then I'm not sure whether lofting can be reliable for mixed profiles.